server/app/assets/javascripts/questionBank.ts

```diff
@@ -134,6 +147,51 @@ class QuestionBankController {
       )
     })
   }
+
+  /**
+   * Sort questions in the question bank based on the criteria selected from the dropdown.
+   */
+  private static sortQuestions() {
+    const questionBankSort = document.getElementById(
+      QuestionBankController.SORT_SELECT_ID,
+    ) as HTMLSelectElement
+
+    const parent = document.querySelector('#non-archived-questions')
+    if (!questionBankSort || !parent) {
+      return
+    }
+
+    const questions: HTMLElement[] = Array.from(
+      parent.querySelectorAll('.cf-question-bank-element'),
+    )
+
+    const sortedQuestions = questions.sort((elementA, elementB) => {
+      const [attrName, order] = questionBankSort.value.split('-')
+      const attrA: string | null = elementA.getAttribute('data-' + attrName)
+      const attrB: string | null = elementB.getAttribute('data-' + attrName)
```

Comment:
Would you add a comment about where data- is coming from, and the expected structure of `questionBankSort.value`?

Reply:
The data prefix is just how data attributes work: https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes. Added comments.